google/cloud/spanner/integration_tests/data_types_integration_test.cc

```diff
@@ -173,22 +173,22 @@ TEST_F(DataTypeIntegrationTest, WriteReadBytes) {
 }
 
 TEST_F(DataTypeIntegrationTest, WriteReadTimestamp) {
-  // TODO(#1098): `Timestamp` cannot represent these extreme values.
-  // auto min = internal::TimestampFromString("0001-01-01T00:00:00Z");
-  // auto max = internal::TimestampFromString("9999-12-31T23:59:59.999999999Z");
-
-  // ASSERT_STATUS_OK(min);
-  // ASSERT_STATUS_OK(max);
+  auto min = Timestamp::FromString("0001-01-01T00:00:00Z");
+  ASSERT_STATUS_OK(min);
+  auto max = Timestamp::FromString("9999-12-31T23:59:59.999999999Z");
+  ASSERT_STATUS_OK(max);
+  auto now = Timestamp::FromChrono(std::chrono::system_clock::now());
```

Comment:
I think this would be the first type in this Spanner API that has a factory function like `Foo::FromBlah()`. In other cases we would make this a non-member factory named `MakeTimestamp()`. What do you think about changing this API to match?

You could also change `Timestamp::ToChrono<T>()` to be `T Timestamp::get<T>()`, which would match what we do in other places, like `Value`, and `Bytes`.

Not that one way is strictly better/worse than the other; my question is simply about the consistency between the various classes in this API.

Thoughts?

Reply:
I was mostly swayed by the internal consistency of closely-declared "static 'From' factory" and "'To'  converter" pairs for each of string/proto/chrono than by any consistency with other classes.

That said, I'm happy to discuss.